lib/event.js

```diff
@@ -15,8 +15,51 @@ class EventModel extends BaseModel {
         super('event', config);
     }
 
+    /**
+     * Return stage builds that belong to this event
+     * @method getStageBuilds
+     * @return {Promise}  Resolves to an array of stage builds
+     */
+    async getStageBuilds() {
+        const nodes = this.workflowGraph.nodes;
+        const stageNamesSet = new Set();
+
+        nodes.forEach(n => {
+            if (n.stageName) {
+                stageNamesSet.add(n.stageName);
+            }
+        });
+
+        const stageNames = Array.from(stageNamesSet);
+        const stageListConfig = {
+            params: {
+                pipelineId: this.pipelineId,
+                name: stageNames
+            }
+        };
+
+        // Lazy load factory dependency to prevent circular dependency issues
+        // https://nodejs.org/api/modules.html#modules_cycles
+        /* eslint-disable global-require */
+        const StageFactory = require('./stageFactory');
+        /* eslint-enable global-require */
+        const stageFactory = StageFactory.getInstance();
+        const stages = await stageFactory.list(stageListConfig);
+        const stageIds = stages.map(s => s.id);
+
+        // Lazy load factory dependency to prevent circular dependency issues
+        // https://nodejs.org/api/modules.html#modules_cycles
+        /* eslint-disable global-require */
+        const StageBuildFactory = require('./stageBuildFactory');
+        /* eslint-enable global-require */
+        const stageBuildFactory = StageBuildFactory.getInstance();
+
+        return stageBuildFactory.list({ params: { eventId: this.id, stageId: stageIds } });
```

Comment:
Filtering by `stageId` may not influence the result. Filtering only by `eventId` should suffice. 
```suggestion
        /* eslint-enable global-require */
        const stageBuildFactory = StageBuildFactory.getInstance();

        return stageBuildFactory.list({ params: { eventId: this.id } });
```

Reply:
updated